vpc/service/validator.go

```diff
@@ -19,9 +21,94 @@ func (vpcService *vpcService) ValidateAllocationParameters(ctx context.Context,
 	defer span.End()
 	log := ctxlogrus.Extract(ctx)
 	ctx = logger.WithLogger(ctx, log)
-	_ = ctx
+	span.AddAttributes(
+		trace.StringAttribute("account", req.AccountId),
+		trace.StringAttribute("subnet-ids", fmt.Sprintf("%v", req.Subnets)),
+		trace.StringAttribute("security-groups", fmt.Sprintf("%v", req.Subnets)),
+	)
+	tx, err := vpcService.db.BeginTx(ctx, &sql.TxOptions{
+		ReadOnly: true,
+	})
+	if err != nil {
+		err = tracehelpers.WithGRPCStatusCode(fmt.Errorf("Could not start database transaction: %w", err), codes.Unknown)
+		tracehelpers.SetStatus(err, span)
+		return nil, err
+	}
 
-	err := status.Error(codes.Unimplemented, "Call not yet implemented")
-	tracehelpers.SetStatus(err, span)
-	return nil, err
+	defer func() {
```

Comment:
@sargun can you explain why we are rolling back? IIUC we are only doing reads currently, and if we ever change this function to modify the DB, this could become a problem?

Reply:
You should always have a rollback statement to avoid leaking transactions. Commit (for writes) should be explicit.